lib/msf/core/exploit/remote/ldap/active_directory.rb

```diff
@@ -253,6 +253,34 @@ def adds_get_domain_info(ldap)
         }
       end
 
+      # Query the LDAP server to retrieve all Certificate Authority (Enterprise CA) servers in the domain.
+      # @param [Net::LDAP::Connection] ldap The LDAP connection to use for querying.
+      # @return [Array<Hash>] An array of hashes, where each hash contains the `:name` and `:dNSHostName` of a CA server.
+      # @rtype [Array<Hash>]
+      def adds_get_ca_servers(ldap)
+        base_dn = "CN=Enrollment Services,CN=Public Key Services,CN=Services,CN=Configuration,#{@base_dn}"
+        filter = '(objectClass=pKIEnrollmentService)'
+        attributes = ['cn', 'dNSHostName']
+        ca_servers = []
+
+        ldap.search(base: base_dn, filter: filter, attributes: attributes) do |entry|
+          ca_servers << {
+            name: entry[:cn]&.first,
+            dNSHostName: entry[:dNSHostName]&.first
+          }
+        end
+
+        if ca_servers.empty?
+          print_warning('No Certificate Authority servers found in LDAP.')
```

Comment:
Worth considering the removal of log statements here up into your module since this is a generic helper